### PR TITLE
Info long code

### PR DESCRIPTION
**Pull request by jwalton3141**
**Beware: R code wrote by Jack.**

Sometimes code in code chunks is too long and protrudes the shaded area.

This PR introduces a check which prints an info message when lines are too long.

This check should produce an info, not a warning, as we don't really care if code is too long in blocks which have `echo = FALSE`.

Alternatively, we could make this a warning and require `purl = FALSE` on chunks to skip.

OR

Someone better at R than me could use `documentation = 1` in the call to `purl` and discard chunks with `echo = FALSE`

---
EXAMPLE UGLY:

image

### CODE REVIEW

File: R/check_code_style.R [hunk 1]
```diff
@@ -48,14 +48,75 @@ check_python_style = function() {
...
+
+    ## Extract code from code chunks
+    rcode = stringr::str_split(
+              knitr::purl(text = contents,
+                          documentation = 0,
+                          quiet = TRUE),
+              "\n")[[1]]
+
+    ## Linewidth of current document
+    ## linewidth = knitr::opts_chunk$get()$linewidth # nolint
```
Comment (jwalton3141) [inline R/check_code_style.R:70]: @csgillespie I wanted to compare linewidth against that of knitr. But this doesn't work as `set_knitr_opts()` isn't called in the checks

**Comment by csgillespie** [comment]
@jwalton3141 Couldn't resist. Now detect when echo is FALSE, and remove as necessary.

Checked against the git, automate, intro and tidy courses. Seems sensible

**Review by csgillespie — APPROVED** [review]
I've made a few changes. Can you double check please

**Comment by jwalton3141** [comment]
I really like the changes to skip `echo = FALSE` chunks.

But this only works for r chunks. For reasons I don't understand purl does not produce the chunk dividers (or grab the chunk options) for non-r chunks, when run with `documentation = 1`.

Without the dividers the `echo = FALSE` logic fails and non-r code is skipped by this check.

**Pull request merged**